Bio/PDB/Mutate.py

```diff
@@ -0,0 +1,578 @@
+                if (
+                    close_atom.get_parent().get_id()[1] == res_num
+                    and chain_if_close_atom.get_id() == chain
+                ):
+                    # print("skipping_own_atom")
+                    continue
+
+                if skip_own_chain and chain_if_close_atom.get_id() == chain:
+                    # print("Skipping own chain")
+                    continue
+                if abs(close_residue.get_id()[1] - res_num) == 1 and is_backbone(
+                    close_atom
+                ):
+                    continue
+                dist = distance(close_atom.coord, rotamer_vector)
+                if dist > 6:
+                    continue
+                try:
+                    vdw_radi = (
+                        VW_RADII[close_atom.get_parent().get_resname()][
+                            close_atom.get_id()
+                        ]
+                        + VW_RADII[mutate_to][rotamer_atom]
+                    )
+                except KeyError:
+                    continue
+                attractive_force = (vdw_radi / dist) ** 6
+                vdw_energy += attractive_force**2 - attractive_force
+        if vdw_energy < lowest_energy:
+            lowest_energy = vdw_energy
+            best_rotamer = rotamer
+    return best_rotamer
+
+
+def mutate(
+    pdb_obj,
+    chain,
+    res_num,
+    mutate_to,
+    rotamer_lib=None,
+    mutation_type="best",
+    verbose="info",
+):
+    """
+    Mutates a given residue based on the best fitting rotamers
+    """
+    level = logging.getLevelName(verbose.upper())
+    logging.getLogger().setLevel(level)
+    Polypeptide.Polypeptide(
+        pdb_obj[0][chain]
+    ).get_phi_psi_list()  # This generates the xtra attribute for PHI and PSI angles for residues
+    try:
+        _residue = pdb_obj[0][chain][res_num]
+    except KeyError:
+        raise KeyError(f"Residue {res_num} not found in chain {chain}!")
+    for atom in list(_residue.get_atoms()):  # Create a copy to remove from
+        if not is_backbone(atom):
+            atom.parent.detach_child(atom.id)
+    phi, psi = (
+        round(np.rad2deg(y), -1) if y else 0
+        for y in [_residue.xtra[x] for x in ["PHI", "PSI"]]
+    )
+    logging.debug(f"Torsion angles: {phi}, {psi}")
+    # GET_ATR IS WRONG
+    sample_residue = read_sample_residue(mutate_to)
+    starting_points = np.asmatrix(
+        [sample_residue["N"], sample_residue["CA"], sample_residue["C"]]
+    )
+    end_points = np.asmatrix(
+        [_residue["N"].coord, _residue["CA"].coord, _residue["C"].coord]
+    )
+
+    sup = SVDSuperimposer.SVDSuperimposer()
+    sup.set(end_points, starting_points)
+    sup.run()
+    rot, tran = sup.get_rotran()
+
+    for atom, coords in sample_residue.items():
+        sample_residue[atom] = np.squeeze(np.asarray(np.dot(coords, rot) + tran))
+
+    if mutate_to not in ["ALA", "GLY"]:
+        if not rotamer_lib:
+            rotamer_lib = load_rotamers()
```

Comment:
I haven't profiled your code in detail, but it seems like loading these 100k lines of data will end up being the main bottleneck for calling this function by default. For example, speeding up that function caused the tests to run about twice as fast. It's good that you included a rotamer_lib option which will allow a knowledgeable user to avoid this problem, but I wonder if there is some way we can make the default a bit faster? 

I guess you could cache the library, but it is pretty big (41MB in memory). Alternatively, I guess you could set it up as a class, and with this as a method, and that way it will got garbage collected when the user is done with it. Drawback of that might be some more complexity, and the user might not realize they should keep around the class

Reply:
Hmm, well, might be worth at least mentioning in the doc string that the user should supply a library to this function if they plan on calling it multiple times